### ISSUE TITLE
File.stat.writable? returning false in JRuby, true in MRI

### ISSUE THREAD

**Issue by duncan-bayne** [jruby/jruby#5641]
### Environment

* `jruby 9.1.7.0 (2.3.1) 2017-01-11 68056ae OpenJDK 64-Bit Server VM 25.192-b26 on 1.8.0_192-b26 +jit [freebsd-x86_64]`
* `FreeBSD x220 12.0-RELEASE FreeBSD 12.0-RELEASE r341666 GENERIC  amd64`

### Expected Behavior

I expect JRuby and MRI Ruby to return the same value - `true` - when `File.stat.writable?` is run on a writable directory (`/tmp`).

### Actual Behavior

JRuby returns false, where MRI returns true:

```
➜  ~ ls -lad /tmp
drwxrwxrwt  21 root  wheel  1024  4 Mar 09:43 /tmp
➜  ~ RBENV_VERSION=2.5.1 irb
irb(main):001:0> File.stat('/tmp').writable?
=> true
```

```
➜  ~ ls -lad /tmp
drwxrwxrwt  21 root  wheel  1024  4 Mar 09:43 /tmp
➜  ~ RBENV_VERSION=jruby-9.1.7.0 irb
irb(main):001:0> File.stat('/tmp').writable?
=> false
```

This is causing many failures, including an inability to install Gems, because `Dir::tmpdir` is failing to find a writable tmpdir.

**Comment by duncan-bayne** [comment]
`truss` suggests the call to `fstatat` is returning the correct permissions:

```
fstatat(AT_FDCWD,"/tmp",{ mode=drwxrwxrwt ,inode=15328896,size=1024,blksize=32768 },0x0) = 0 (0x0)
```

**Comment by Freaky** [comment]
See #5521 - until https://github.com/jnr/jnr-posix/issues/126 is fixed, you'll want to run JRuby under an 11.2 chroot or jail so it sees the old stat structure.

**Comment by headius** [comment]
@Freaky Sorry that jnr/jnr-posix#126 got left behind. Let's get it merged and fix this!

**Comment by enebo** [comment]
jnr-posix was updated so this is fixed.

**Comment by Freaky** [comment]
Hmm, just ran a build off latest jruby-master, and I'm *not* seeing this issue resolved despite it apparently using jnr-posix 3.0.50 which should have my fix.

I'd be expecting it to call out to `/bin/freebsd-version` to determine the appropriate ABI, but I'm not seeing that with ktrace...

**Comment by Freaky** [comment]
Disregard that, I was holding it wrong. Ahem.
